### PR TITLE
Fixes 2x data in report generation logic

### PR DESCRIPTION
**Pull request by danishjamal104**
Updates in this PR
1. Added Payment mode configuration in `application.yaml`.
2. Updated the report generation to fix the 2x data calculation for batch summary.
3. Added `lambok` for auto-generation of boilerplate code in configuration POJO.

### CODE REVIEW

**Comment by danishjamal104** [comment]
@avikganguly01 and @fynmanoj please review and merge. Thanks :)

File: src/main/java/org/apache/fineract/config/PaymentModeConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+@Setter
+@NoArgsConstructor
```
Comment (fynmanoj) [inline src/main/java/org/apache/fineract/config/PaymentModeConfiguration.java:13]: we can skip @NoArgsConstructor if there is no explicit constructor,   NoArgsConstructor is a java default behaviour

File: src/main/java/org/apache/fineract/operations/VariableRepository.java [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+import org.springframework.data.repository.query.Param;
 
 import java.util.List;
 
 public interface VariableRepository extends CrudRepository<Variable, Long> {
 
     List<Variable> findByWorkflowInstanceKeyOrderByTimestamp(Long workflowInstanceKey);
 
+    @Query("SELECT v from Variable v WHERE v.workflowInstanceKey=:workflowInstanceKey and v.name=:name")
+    Variable findByWorkflowInstanceKeyAndVariableName(@Param("name") String name,
```
Comment (fynmanoj) [inline src/main/java/org/apache/fineract/operations/VariableRepository.java:14]: use Optional here

File: src/main/java/org/apache/fineract/api/BatchApi.java [hunk 3]
```diff
@@ -143,6 +151,15 @@ private BatchDTO generateDetails (Batch batch) {
         BigDecimal failedAmount = BigDecimal.ZERO;
 
         for (Transfer transfer : transfers) {
+            Variable variable = variableRepository.findByWorkflowInstanceKeyAndVariableName("paymentMode",
```
Comment (fynmanoj) [inline src/main/java/org/apache/fineract/api/BatchApi.java:154]: use `Optional<Variable>` and `isPresent`

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/fineract/config/PaymentModeConfiguration.java (hunk 1 above)
Comment (danishjamal104) [reply src/main/java/org/apache/fineract/config/PaymentModeConfiguration.java:13]: it gives me error. Saying no empty constructor found.

File: src/main/java/org/apache/fineract/operations/VariableRepository.java (hunk 2 above)
Comment (danishjamal104) [reply src/main/java/org/apache/fineract/operations/VariableRepository.java]: done

File: src/main/java/org/apache/fineract/api/BatchApi.java (hunk 3 above)
Comment (danishjamal104) [reply src/main/java/org/apache/fineract/api/BatchApi.java]: done

File: src/main/java/org/apache/fineract/config/PaymentModeConfiguration.java (hunk 1 above)
Comment (fynmanoj) [reply src/main/java/org/apache/fineract/config/PaymentModeConfiguration.java:13]: Need to check this. It should add it by default

**Pull request merged**